mmdet3d/models/backbones/cylinder3d.py

```diff
@@ -0,0 +1,595 @@
+              indice_key: Optional[str] = None) -> SparseModule:
+    """SubMConv3d with kernel size of 1*3*1.
+
+    Args:
+        in_channels (int): Input channels of the block.
+        out_channels (int): Output channels of the block.
+        stride (int): Conv stride. Defaults to 1.
+        indice_key (str, optional): Name of indice tables. Defaults to None.
+
+    Returns:
+        SparseModule: SubMConv3d with kernel size of 1*3*1.
+    """
+    return SubMConv3d(
+        in_channels,
+        out_channels,
+        kernel_size=(1, 3, 1),
+        stride=stride,
+        padding=(0, 1, 0),
+        bias=False,
+        indice_key=indice_key)
+
+
+def conv3x1x1(in_channels: int,
+              out_channels: int,
+              stride: int = 1,
+              indice_key: Optional[str] = None) -> SparseModule:
+    """SubMConv3d with kernel size of 3*1*1.
+
+    Args:
+        in_channels (int): Input channels of the block.
+        out_channels (int): Output channels of the block.
+        stride (int): Conv stride. Defaults to 1.
+        indice_key (str, optional): Name of indice tables. Defaults to None.
+
+    Returns:
+        SparseModule: SubMConv3d with kernel size of 3*1*1.
+    """
+    return SubMConv3d(
+        in_channels,
+        out_channels,
+        kernel_size=(3, 1, 1),
+        stride=stride,
+        padding=(1, 0, 0),
+        bias=False,
+        indice_key=indice_key)
+
+
+def conv3x1x3(in_channels: int,
+              out_channels: int,
+              stride: int = 1,
+              indice_key: Optional[str] = None) -> SparseModule:
+    """SubMConv3d with kernel size of 3*1*3.
+
+    Args:
+        in_channels (int): Input channels of the block.
+        out_channels (int): Output channels of the block.
+        stride (int): Conv stride. Defaults to 1.
+        indice_key (str, optional): Name of indice tables. Defaults to None.
+
+    Returns:
+        SparseModule: SubMConv3d with kernel size of 3*1*3.
+    """
+    return SubMConv3d(
+        in_channels,
+        out_channels,
+        kernel_size=(3, 1, 3),
+        stride=stride,
+        padding=(1, 0, 1),
+        bias=False,
+        indice_key=indice_key)
+
+
+def conv1x1x1(in_channels: int,
+              out_channels: int,
+              stride: int = 1,
+              indice_key: Optional[str] = None) -> SparseModule:
+    """SubMConv3d with kernel size of 1*1*1.
+
+    Args:
+        in_channels (int): Input channels of the block.
+        out_channels (int): Output channels of the block.
+        stride (int): Conv stride. Defaults to 1.
+        indice_key (str, optional): Name of indice tables. Defaults to None.
+
+    Returns:
+        SparseModule: SubMConv3d with kernel size of 1*1*1.
+    """
+    return SubMConv3d(
+        in_channels,
+        out_channels,
+        kernel_size=1,
+        stride=stride,
+        padding=1,
+        bias=False,
+        indice_key=indice_key)
+
+
+class AsymmResBlock(BaseModule):
```

Comment:
I suggest putting these blocks into sparse_block.py so that other methods can use them.